TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs

```diff
@@ -117,22 +125,28 @@ public TimedTrafficLightsTool(TrafficManagerTool mainTool)
             MainTool.RequestOnscreenDisplayUpdate();
         }
 
-        public override void OnSecondaryClickOverlay() {
-            if (!IsCursorInPanel()) {
-                if (ttlToolMode_ != TTLToolMode.SelectNode) {
-                    Cleanup();
-                    SetToolMode(TTLToolMode.SelectNode);
-                } else {
-                    MainTool.SetToolMode(ToolMode.None);
-                }
+        public override void OnToolRightClick() {
+            if (IsCursorInPanel()) {
+                return;
             }
+
+            if (ttlToolMode_ != TtlToolMode.SelectNode) {
+                OnDeactivateTool();
+                SetToolMode(TtlToolMode.SelectNode);
+            } else {
+                MainTool.SetToolMode(ToolMode.None);
+            }
+        }
+
+        public override void UpdateEveryFrame() {
+            throw new NotImplementedException();
```

Comment:
is this going to throw exception every frame?

Reply:
```cs
public void OnToolGUIImpl(Event e) {
	...
	if (activeLegacySubTool_ != null) {
		activeLegacySubTool_.OnToolGUI(e);
	} else if (activeSubTool_ != null) {
		activeSubTool_.UpdateEveryFrame();
	}
```            
I feel like `UpdateEveryFrame` is called multiple times per frame: several times in Update phase and several more times in OnGUI phase.